plugin/pkg/scheduler/schedulercache/interface.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schedulercache
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+// Cache collects pods' information and provides node-level aggregated information.
+// It's intended for generic scheduler to do efficient lookup.
+// Cache's operations are pod centric. It incrementally updates itself based on pod event.
+// Pod events are sent via network. We don't have guaranteed delivery of all events.
+//
+// State Machine of a pod's events in scheduler's cache:
+//
+//                                                +-------+
+//                                                |       |
+//                                                |       | Update
+//           Assume                Add            +       |
+// Initial +--------> Assumed +------------+---> Added <--+
+//                       +                 |       +
+//                       |                 |       |
+//                       |             Add |       | Remove
+//                       |                 |       |
+//                       |                 +       |
+//                       +-------------> Expired   +----> Deleted
+//                          expire
+//
+// Note that an assumed pod can expire. Because if we haven't received Add event notifying us
+// for a while, there might be some problems and we shouldn't keep the pod in cache anymore.
+//
+// Note that "Initial", "Expired", and "Deleted" pods do not actually exist in cache.
+// Based on existing use cases, we are making following assumptions:
+// - No pod would be assumed twice
+// - If a pod wasn't added, it wouldn't be removed or updated.
+// - Both "Expired" and "Deleted" are valid end states. If network disconnection occurred,
+// 	 a pod could have been added and deleted from api server and local cache knew nothing.
```

Comment:
Sorry for being boring about this part, but I'm afraid I still don't really follow this comment (although I think I understand what you wanted to write).
Maybe replace the second sentence with something like:
"In case of some problems, e.g. network issue, a pod might have changed its state (e.g. been deleted) without delivering notification to the cache."
WDYT?